### PR TITLE
810-2fa-doc

### LINKED ISSUE #810: 2fa documentation
rearranging security folder
adding document: 2fa

### PR DESCRIPTION
**Pull request by eeliang**
fixes #810 

NEW:
security/2fa.md
assets/images/security/2fa/

EDITS:
SUMMARY.md
security.md

Moved:
security.md#ip-whitelists > security/ip-whitelists.md

### CODE REVIEW

File: security/2fa.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+![Workato requesting two-factor authentication](/assets/images/security/2fa/2fa-request.png)
+*Workato requesting two-factor authentication*
+
+A unique verification code is generated for you whenever you login to your Workato account. It is a temporary 6-digit code that lets us know it is indeed *you*. Afterwards, the code will expire and cannot be reused.
+
+Workato allows authentication through mobile applications and hardware security keys. Some of the common ones are:
+- [Authy](https://authy.com/download/)
+- [Symantec VIP](https://vip.symantec.com/)
+- [Yubico](https://www.yubico.com/products/yubikey-hardware/) (hardware security key)
```
Comment (kandadaboggu) [inline security/2fa.md]: Do we know if this works?

File: security/2fa.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+### Setup two-factor authentication
+
+Let's go through the activation for two-factor authentication. You will need to be logged in to your Workato account and have your [authenticator app](https://www.pcworld.com/article/3225913/what-is-two-factor-authentication-and-which-2fa-apps-are-best.html) ready. In this example, we will be using **Symantec VIP**.
```
Comment (kandadaboggu) [inline security/2fa.md]: Use Google authenticator as example.

File: security/2fa.md [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+
+<table class="unchanged rich-diff-level-one">
```
Comment (kandadaboggu) [inline security/2fa.md]: why is this an HTML table? Why not use markdown table?

File: security/2fa.md [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+If you realised that you have misplaced your recovery codes, we advise that you generate a new set. Remember that other people can use the recovery codes to access to your Workato account!
+
+You can also generate new recovery codes if they are depleting. Using the **single-use** recovery codes for two-factor authentication will reduce the total number of available codes. Generating a new set of recovery codes will ensure that you never get locked out of your Workato account.
+
+Select **Account settings** > **Two-factor authentication** > **View recovery codes** > **Regenerate recovery codes**
+
+![Generate new recovery codes](/assets/images/security/2fa/2fa-regenerate-codes.png)
+*Generate new recovery codes*
+
+#### Trusted devices
```
Comment (kandadaboggu) [inline security/2fa.md]: Add a section about `sudo` mode, that requires user to enter password again to make changes to 2fa configuration.

File: security/2fa.md [hunk 5]
```diff
@@ -0,0 +1,177 @@
+---
+title: Two-factor authentication
```
Comment (kandadaboggu) [inline security/2fa.md]: Make sure we use the same url in the product code.

File: security/2fa.md (hunk 1 above)
Comment (eeliang) [reply security/2fa.md]: I have tested the mobile apps and they work, including google authenticator. I have not tested the hardware security key

File: security/2fa.md (hunk 3 above)
Comment (eeliang) [reply security/2fa.md]: We are using multi-line text and images in this table. HTML makes the code more readable

File: SUMMARY.md [hunk 6]
```diff
@@ -580,6 +580,7 @@
     * [Users](workato-api/users.md)
     * [Recipe lifecycle management](workato-api/recipe-lifecycle-management.md)
 * [Security](security.md)
-  * [IP whitelists](security.md#ip_whitelists)
+  * [IP whitelists](/security/ip-whitelists.md)
```
Comment (eeshansim) [inline SUMMARY.md]: keep to the convention for this doc and use relative path

```markdown
* [IP whitelists](security/ip-whitelists.md)
```

**Pull request merged**